nebula_common/include/nebula_common/continental/continental_srr520.hpp

```diff
@@ -0,0 +1,295 @@
+{
+  uint8_t data[DETECTION_FRAGMENT_SIZE];
+};
+
+struct DetectionPacket
+{
+  DetectionFragmentPacket fragments[FRAGMENTS_PER_DETECTION_PACKET];  // 0 - 59
+  uint8_t reserved0;                                                  // 60
+  uint8_t reserved1;                                                  // 61
+  uint8_t u_message_counter;                                          // 62
+  uint8_t u_sequence_counter;                                         // 63
+};
+
+struct ObjectHeaderPacket
+{
+  big_uint32_buf_t u_time_stamp;              // 0
+  big_uint32_buf_t u_global_time_stamp_sec;   // 4
+  big_uint32_buf_t u_global_time_stamp_nsec;  // 8
+  uint8_t u_global_time_stamp_sync_status;    // 12
+  uint8_t u_signal_status;                    // 13
+  uint8_t u_sequence_counter;                 // 14
+  big_uint32_buf_t u_cycle_counter;           // 15
+  big_uint16_buf_t u_ego_vx;                  // 19
+  big_uint16_buf_t u_ego_yaw_rate;            // 21
+  uint8_t u_motion_type;                      // 23
+  uint8_t u_number_of_objects;                // 24
+  uint8_t reserved[7];                        // 25
+};
+
+struct ObjectFragmentPacket
+{
+  uint8_t data[OBJECT_FRAGMENT_SIZE];
+};
+
+struct ObjectPacket
+{
+  ObjectFragmentPacket fragments[FRAGMENTS_PER_OBJECT_PACKET];  // 0 - 61
+  uint8_t u_message_counter;                                    // 62
+  uint8_t u_sequence_counter;                                   // 63
+};
+
+struct ConfigurationPacket
+{
+  uint8_t u_sensor_id;
+  big_uint16_buf_t u_long_pose;
+  big_uint16_buf_t u_lat_pose;
+  big_uint16_buf_t u_vert_pos;
+  big_uint16_buf_t u_long_pos_cog;
+  big_uint16_buf_t u_wheelbase;
+  big_uint16_buf_t u_yaw_angle;
+  big_uint16_buf_t u_cover_damping;
+  uint8_t u_plug_orientation_and_default_settings;
+};
+
+struct SyncPacket
+{
+  uint8_t u_type;
+  uint8_t u_crc;
+  uint8_t u_time_domain_and_sequence_counter;
+  uint8_t u_user_data0;
+  big_uint32_buf_t u_sync_time_sec;
+};
+
+struct FollowUpPacket
+{
+  uint8_t u_type;
+  uint8_t u_crc;
+  uint8_t u_time_domain_and_sequence_counter;
+  uint8_t u_reserved;
+  big_uint32_buf_t u_sync_time_nsec;
+};
+
+#pragma pack(pop)
+
+struct PointSRR520Detection
+{
+  PCL_ADD_POINT4D;
+  float range;
+  float azimuth;
+  float range_rate;
+  float rcs;
+  uint8_t pdh00;
+  uint8_t pdh01;
+  uint8_t pdh02;
+  uint8_t pdh03;
+  uint8_t pdh04;
+  uint8_t pdh05;
+  float snr;
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+} EIGEN_ALIGN16;
+
+// Note we only use a subset of the data since POINT_CLOUD_REGISTER_POINT_STRUCT has a limit in the
+// number of fields
+struct PointSRR520Object
+{
+  PCL_ADD_POINT4D;
+  uint32_t id;
+  uint16_t age;
+  float orientation;
+  float rcs;
+  float score;
+  uint8_t object_status;
+  float dynamics_abs_vel_x;
+  float dynamics_abs_vel_y;
+  float dynamics_abs_acc_x;
+  float dynamics_abs_acc_y;
+  float box_length;
+  float box_width;
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+} EIGEN_ALIGN16;
```

Comment:
Please move `EIGEN_ALIGN16` from behind the struct to between the `struct` keyword and name to stay Clang-compatible. See [PCL#3237](https://github.com/PointCloudLibrary/pcl/pull/3237) for more info.

Reply:
Addressed in [d545d84](https://github.com/tier4/nebula/pull/124/commits/d545d84b033550f6ae16b2aed141f18fdb24ffd0)